sched/environ/env_unsetenv.c

```diff
@@ -61,23 +61,30 @@ int unsetenv(FAR const char *name)
 {
   FAR struct tcb_s *rtcb = this_task();
   FAR struct task_group_s *group = rtcb->group;
-  int ret = OK;
+  int idx;
 
-  DEBUGASSERT(name && group);
+  DEBUGASSERT(group);
+
+  /* Check the incoming parameter */
+
+  if (name == NULL || *name == '\0' || strchr (name, '=') != NULL)
```

Comment:
```suggestion
  DEBUGASSERT(name && group);

  /* Check the incoming parameter */

  if (*name == '\0' || strchr (name, '=') != NULL)
```

Reply:
My thought was not to crash the whole system here but I can revert to the old implementation too.

Reply:
I'm not sure. We can leave `NULL` check as `EINVAL`. That makes sense. I just was thinking that POSIX states nothing about NULL case, but only covers empty string and `'='` holder string. I'm fine to keep the `NULL` check as recoverable error. Not sure how Linux manages that.

Reply:
Yes glibc tests for the NULL pointer